src/Icons/src/Twig/UXIconComponentListener.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Twig;
+
+use Symfony\UX\Icons\IconRenderer;
+use Symfony\UX\TwigComponent\Event\PreCreateForRenderEvent;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @internal
+ */
+final class UXIconComponentListener
+{
+    public function __construct(
+        private IconRenderer $iconRenderer,
+    ) {
+    }
+
+    public function onPreCreateForRender(PreCreateForRenderEvent $event): void
+    {
+        if ('UX:Icon' !== $event->getName()) {
```

Comment:
There's not an instance check we can do here? Or even a check for a template? The point of this listener, if I understand correctly, is performance: to short-circuit the rendering process by avoiding the template render and, instead rendering right here, early. Would it make sense to include a nonsense template name and we look for that here? The philosophical idea is that this listener is doing the work of a template - almost standing in for a template. So maybe a template name is what we should use as the key?

Reply:
No it replaces a TwigComponent rendering, so the name is the name of the component :) 

We bypass the TwigComponent instanciation/mount/etc entirely